ssl/quic/quic_txp.c

```diff
@@ -843,11 +843,39 @@ int ossl_quic_tx_packetiser_generate(OSSL_QUIC_TX_PACKETISER *txp,
             }
 
         if (first_el != QUIC_ENC_LEVEL_NUM
-            && total_dgram_size < min_dpl) {
+            && total_dgram_size < min_dpl
+            /*
+             * We might not be able to add padding, for example if we are using
+             * the ACK_ONLY archetype.
+             */
+            && pkt[first_el].geom.adata.allow_padding) {
             size_t deficit = min_dpl - total_dgram_size;
 
+            if (!ossl_assert(!pkt[first_el].h.done_implicit))
```

Comment:
Hmmm.......couldn't the force_pad check mean we have forced explicit encoding for something other than the first_el....i.e. couldn't done_implicit still be set here? i.e. the packet where we have ensured explicit length and the packet where we are adding padding may not be the same one?

Reply:
Hmm, good point.